src/librustdoc/doctest.rs

```diff
@@ -749,28 +755,27 @@ impl Tester for Collector {
             _ => PathBuf::from(r"doctest.rs"),
         };
 
+        // For example `module/file.rs` would become `module_file_rs`
+        let test_id = filename
+            .to_string()
+            .chars()
+            .map(|c| if c.is_ascii_alphanumeric() { c } else { '_' })
+            .collect::<String>();
+        let test_id = format!(
+            "{file}_{line}_{number}",
```

Comment:
This basically reverts half of https://github.com/rust-lang/rust/pull/79413 since https://github.com/rust-lang/cargo/pull/8954 is a proper solution to this. CC @jyn514

Reply:
Sure, seems reasonable. I always want to do things through cargo instead of rustdoc where possible because cargo has more information.